src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -234,120 +249,116 @@ public void update() {
 
-                    itemOut.stackSize = 1;
-                    itemIn.stackSize = itemIn.stackSize - 1;
-                }
+            if (itemIn.stackSize - 1 <= 0) {
+                itemOut = itemIn.copy();
+                itemIn = null;
+            } else {
+                itemOut = itemIn.copy();
 
-                if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
-                if (itemOut.stackSize == 0) itemOut = null;
+                itemOut.stackSize = 1;
+                itemIn.stackSize = itemIn.stackSize - 1;
+            }
 
-                inventory.setInventorySlotContents(0, itemIn);
-                inventory.setInventorySlotContents(1, itemOut);
+            if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
+            if (itemOut.stackSize == 0) itemOut = null;
 
-                ticksRemaining = 200;
-                machineActive = true;
+            inventory.setInventorySlotContents(0, itemIn);
+            inventory.setInventorySlotContents(1, itemOut);
 
-                this.markForUpdate();
-                this.markDirty();
-            }
+            ticksRemaining = 200;
+            machineActive = true;
 
+            this.markForUpdate();
+            this.markDirty();
+        }
 
-            if (fuelRemaining == 0 && machineActive && machineTier == machineTier.TIER_0) {
-                ticksRemaining = 0;
-                test = false;
 
-            } else {
-                test = false;
-            }
+        if (ticksRemaining <= 0 && machineActive) {
 
 
-            if (ticksRemaining <= 0 && machineActive) {
+            ticksRemaining = 0;
 
-                if (test == false) {
-                    System.out.println(test);
-                    ticksRemaining = 0;
+            if (worldObj.isRemote)
+                return;
 
-                    if (worldObj.isRemote)
-                        return;
+            ItemStack processItem = inventory.getStackInSlot(1);
+            if (processItem == null) {
+                return;
+            }
 
-                    ItemStack processItem = inventory.getStackInSlot(1);
-                    if (processItem == null) {
-                        return;
-                    }
+            List<Crushable> outputs = PulverizerRegistry.getOutputs(processItem);
 
-                    List<Crushable> outputs = PulverizerRegistry.getOutputs(processItem);
+            if (outputs.isEmpty())
+                return;
 
-                    if (outputs.isEmpty())
-                        return;
+            for (int i = this.crushIndex; i < outputs.size(); i++) {
+                this.crushIndex = i;
+                Crushable crushable = outputs.get(this.crushIndex);
 
-                    for (int i = this.crushIndex; i < outputs.size(); i++) {
-                        this.crushIndex = i;
-                        Crushable crushable = outputs.get(this.crushIndex);
+                ItemStack outItem = crushable.output.copy();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00122)